File: src/project_manager_flutter/lib/features/clients/models/client_contact.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'client_contact.freezed.dart';
part 'client_contact.g.dart';

@freezed
abstract class ClientContact with _$ClientContact {
  const factory ClientContact({
    required String id,
    required String name,
    String? email,
    String? phone,
    required bool isStakeHolder,
    required bool isInvoicing,
    required DateTime createdAt,
  }) = _ClientContact;

  factory ClientContact.fromJson(Map<String, dynamic> json) =>
      _$ClientContactFromJson(json);
}

File: src/ProjectManager.Api/Data/Entities/ClientContact.cs
namespace ProjectManager.Api.Data.Entities;

public class ClientContact
{
    public Guid Id { get; set; }
    public Guid ClientId { get; set; }
    public Client Client { get; set; } = null!;
    public string Name { get; set; } = string.Empty;
    public string? Email { get; set; }
    public string? Phone { get; set; }
    public bool IsStakeHolder { get; set; }
    public bool IsInvoicing { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
}
